modules/monitoring-supported-remote-write-authentication-settings.adoc

```diff
@@ -0,0 +1,331 @@
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: REFERENCE
+[id="supported_remote_write_authentication_settings_{context}"]
+= Supported remote write authentication settings
+
+You can use different methods to authenticate with a remote write endpoint.
+Currently supported authentication methods are AWS Signature Version 4, basic authentication, bearer token authorization, OAuth 2.0, and TLS client.
+The following table provides details about supported authentication methods for use with remote write.
+
+[options="header"]
+|===
+
+|Authentication method|Config map field|Description
+
+|AWS Signature Version 4|`sigv4`|This method uses AWS Signature Version 4 authentication to sign requests.
+You cannot use this method simultaneously with bearer token, OAuth 2.0, or basic authentication.
+
+|basic authentication|`basicAuth`|Basic authentication sets the authorization header on every remote write request with the configured username and password.
+
+|bearer token authorization|`Bearer` or `bearerFile`|Bearer token authentication sets the authorization header on every remote write request using the configured token.
+You can configure bearer token authentication in one of two ways: by pointing to a bearer token file in the remote write `bearerTokenFile` field or by specifying a secret that contains the bearer token information. 
+In the second method, the `remoteWrite` settings in the config map must include an `authorization` `type` of `Bearer`.
+
+|OAuth 2.0|`oauth2`|An OAuth 2.0 configuration uses the client credentials grant type. 
+Prometheus fetches an access token from `tokenUrl` with the specified client ID and client secret to access the remote write endpoint using this token.
+You cannot use this method simultaneously with bearer token, AWS Signature Version 4, or basic authentication.
+
+|TLS client|`tlsConfig`|A TLS client configuration specifies the CA certificate, the client certificate, and the client key file information used to authenticate with the remote write endpoint server using TLS.
+The sample configuration assumes that you have already created a CA certificate file, a client certificate file, and a client key file.
+
+|===
+
+== Config map location for authentication settings
+The following shows the location of the authentication configuration in the `ConfigMap` object.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote-write-endpoint.example.com" <1>
+        <endpoint_authentication_details> <2>
+----
+<1> The URL of the remote write endpoint.
+<2> The required configuration details for the authentication method for the endpoint.
+Currently supported authentication methods are Amazon Web Services (AWS) Signature Version 4, bearer token, basic authentication, OAuth 2.0, and TLS client. 
```

Comment:
Maybe here it also makes sense to add the note about the UWM ConfigMap just for good measure, WDYT?

Reply:
Thanks @JoaoBraveCoding. That's a good idea. I've added the following admonition here:

```
[NOTE]
====
If you want to configure remote write for the Prometheus instance that monitors user-defined projects, edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` namespace. Note that TLS client configuration for remote write is not supported for use with user-defined projects.
====
```